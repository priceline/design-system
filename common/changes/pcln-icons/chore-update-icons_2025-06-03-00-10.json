{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "Update to styled-system v5 on pcln-icons, move to dev dependency and peer dependency. Use Node v20 for the build process.",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}