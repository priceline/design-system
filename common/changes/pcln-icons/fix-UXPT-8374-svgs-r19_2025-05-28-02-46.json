{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "replace defaultProps usage in svgr template file",
      "type": "patch"
    }
  ],
  "packageName": "pcln-icons"
}