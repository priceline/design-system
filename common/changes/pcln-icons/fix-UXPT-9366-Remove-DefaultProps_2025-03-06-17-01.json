{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "remove defaultProps",
      "type": "minor"
    }
  ],
  "packageName": "pcln-icons"
}