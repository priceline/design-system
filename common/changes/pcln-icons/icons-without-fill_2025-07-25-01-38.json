{
  "changes": [
    {
      "packageName": "pcln-icons",
      "comment": "UXPT-10383: rounded Material SVGs from filled to unfilled",
      "type": "major"
    }
  ],
  "packageName": "pcln-icons"
}