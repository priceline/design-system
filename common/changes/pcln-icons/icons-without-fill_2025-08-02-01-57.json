{
  "changes": [
    {
      "comment": "allow styled-components 6 as a peer",
      "type": "patch",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "steven.dalonzo@priceline.com"
}