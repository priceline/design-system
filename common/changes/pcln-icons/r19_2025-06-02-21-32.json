{
  "changes": [
    {
      "comment": "React 19 ecosystem upgrade",
      "type": "major",
      "packageName": "pcln-icons"
    }
  ],
  "packageName": "pcln-icons",
  "email": "steven.dalonzo@priceline.com"
}