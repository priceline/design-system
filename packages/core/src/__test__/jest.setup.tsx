import '@testing-library/jest-dom'
import 'jest-styled-components'

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
