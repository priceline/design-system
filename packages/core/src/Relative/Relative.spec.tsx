import { Coupon as CouponIcon } from 'pcln-icons'
import { render, screen } from '../__test__/testing-library'
import { Absolute } from '../Absolute/Absolute'
import { Relative } from '../Relative/Relative'
import { Text } from '../Text/Text'

describe('Relative', () => {
  test('renders with top, left and zIndex props', () => {
    render(<Relative top={10} left={0} zIndex={2} data-testid='relative' />)
    const element = screen.getByTestId('relative')
    expect(element).toHaveStyleRule('position', 'relative')
    expect(element).toHaveStyleRule('top', '10px')
    expect(element).toHaveStyleRule('left', '0px')
    expect(element).toHaveStyleRule('z-index', '2')
  })

  test('renders with flag and Child', () => {
    const { asFragment } = render(
      <Relative top={10} left={0}>
        <Absolute top={10} right={0} zIndex={2}>
          <CouponIcon />
          <Text.span>EXCLUSIVE</Text.span>
        </Absolute>
      </Relative>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
