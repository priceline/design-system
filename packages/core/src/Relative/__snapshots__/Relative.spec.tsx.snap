// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Relative renders with flag and Child 1`] = `
<DocumentFragment>
  .c3 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  line-height: 1;
  width: 24px;
}

.c4 {
  outline: none;
}

.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c2 {
  position: absolute;
  top: 10px;
  right: 0px;
  z-index: 2;
}

.c1 {
  position: relative;
  top: 10px;
  left: 0px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <svg
          aria-hidden="true"
          class="c3 c4"
          fill="currentcolor"
          focusable="false"
          height="24"
          role="img"
          size="24"
          tabindex="-1"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.4 11.6l-9-9c-.3-.4-.8-.6-1.4-.6H4c-1.1 0-2 .9-2 2v7c0 .6.2 1.1.6 1.4l9 9c.4.4.9.6 1.4.6s1.1-.2 1.4-.6l7-7c.4-.4.6-.9.6-1.4s-.2-1.1-.6-1.4zM5.5 7C4.7 7 4 6.3 4 5.5S4.7 4 5.5 4 7 4.7 7 5.5 6.3 7 5.5 7z"
          />
        </svg>
        <span
          class=""
        >
          EXCLUSIVE
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
